app/lib/ldap_authenticatable_tiny/strategy.rb

```diff
@@ -14,10 +14,13 @@ class LdapAuthenticatableTiny < Authenticatable
       # https://github.com/plataformatec/devise/blob/master/lib/devise/models/authenticatable.rb
       def authenticate!
         email = authentication_hash[:email]
+        should_use_mock_ldap = ::EnvironmentVariable.is_true('USE_TEST_PALS_LDAP')
```

Comment:
Adding other guards here seems good, like disallowing this in production or only allowing it when the district_key is `demo`.

Reply:
I have another guard in the TestPalsMockLDAP class itself. Unfortunately I don't think we can disallow this for `Rails.env.production == true` because the demo Heroku site is production from a Rails.env point of view.

Reply:
Gotcha.  I think it's important to make it so that it takes more steps and efforts than flipping a switch to disable authorization.  So maybe we can find some other ways to do that!

Here's some brainstorming ideas of things we could try:
- Move this check to an initializer, so that we can check all these things at app start time (and crash the app) rather than not catching a bad config until someone tries to sign in.  In the initializer, only allow it to be run in production when it's the `demo` district and when the environment variable is set.  Add that initializer file to mention bot.
- Remove the demo passwords from code and put them in an env variable.  This would require two env variable changes rather than just one.
- Make the demo passwords env variable a map of `email => password` so that even it were copy-pasted into production by accident it wouldn't work

Reply:
I ended up implementing 2/3 of these suggestions:

* We have not one, not two, but three pieces of ENV required to disable real authentication and use the mock LDAP instead. `USE_TEST_PALS_LDAP` and `TEST_PALS_LDAP` password both need to be set and `DISTRICT_KEY` needs to be set to demo. 
* Also added an initializer to run checks to verify sane config and set good defaults on app start. 

I think this is a fairly decent improvement on the status quo, which all forks on a single ENV variable, `SHOULD_USE_LDAP`.

I didn't implement the map of `email => password` suggestion because if the Test Pals ever change down the line, we'd have to remember to replace the ENV variables everywhere and that seemed like something that could really easily fall out of sync.